pkg/pipeline/service.go

```diff
@@ -206,6 +206,11 @@ func (s *Service) AddConnector(ctx context.Context, pipelineID string, connector
 	if err != nil {
 		return nil, err
 	}
+	for _, id := range pl.ConnectorIDs {
```

Comment:
At this point we already created the pipeline and at least 2 connectors before we spotted the issue. The next thing that happens is we roll back the entities that were already provisioned. I'm trying to say we could spot the issue much earlier if we validate that connector IDs are unique when [validating the config](https://github.com/ConduitIO/conduit/blob/03983e01e31990ad6be3019d22d67d0d1ecd0d5f/pkg/provisioning/config/validate.go#L52), then we won't even try to provision the pipeline.

BTW, we should probably also validate that processor IDs are unique ([here](https://github.com/ConduitIO/conduit/blob/03983e01e31990ad6be3019d22d67d0d1ecd0d5f/pkg/provisioning/config/validate.go#L74)), I suspect that having duplicated processor IDs results in a similar panic or unexpected behavior.